sklearn/ensemble/gradient_boosting.py

```diff
@@ -1146,7 +1146,8 @@ def feature_importances_(self):
 
     def _validate_y(self, y):
         self.n_classes_ = 1
-
+        if y.dtype is np.dtype(object):
```

Comment:
If this logic is present in a classifier, it is problematic; strings should be valid as dtype object (e.g. this is default for a pandas.Series of string). If it's only relevant to regressors, it should probably not be in `Base`.